src/mpi/coll/iallreduce/iallreduce_tsp_recexch.c

```diff
@@ -99,7 +100,11 @@ int MPIR_TSP_Iallreduce_sched_intra_recexch(const void *sendbuf, void *recvbuf,
     for (phase = 0; phase < step2_nphases && step1_sendto == -1; phase++) {
         if (!is_commutative) {
             /* sort the neighbor list so that receives can be posted in order */
+#if defined(HAVE_QSORT)
             qsort(step2_nbrs[phase], k - 1, sizeof(int), MPII_Algo_compare_int);
+#else
+            MPIR_insertion_sort(step2_nbrs[phase], k - 1);
+#endif
```

Comment:
We should move the whole thing into `MPL`, call it `MPL_sort_int_array`. The algorithm used for sort is not important here and should be abstracted away inside MPL.

Reply:
done

Reply:
done